lib/js-file.js

```diff
@@ -244,26 +185,60 @@ JsFile.prototype = {
     getLastNodeToken: function(node) {
         return this.getTokenByRangeEnd(node.range[1]);
     },
+
     /**
      * Returns the first token before the given.
      *
      * @param {Object} token
+     * @param {Object} [options]
+     * @param {Boolean} [options.includeComments=false]
      * @returns {Object|undefined}
      */
-    getPrevToken: function(token) {
+    getPrevToken: function(token, options) {
         var index = token._tokenIndex - 1;
-        return index >= 0 ? this._tree.tokens[index] : undefined;
+        if (index < 0) {
+            return undefined;
+        }
+
+        if (options && options.includeComments) {
+            return this._tokens[index];
+        }
+
+        do {
+            if (!this._tokens[index].isComment) {
+                return this._tokens[index];
+            }
+        } while (--index >= 0);
+
+        return undefined;
     },
+
     /**
      * Returns the first token after the given.
      *
      * @param {Object} token
+     * @param {Object} [options]
+     * @param {Boolean} [options.includeComments=false]
      * @returns {Object|undefined}
      */
-    getNextToken: function(token) {
+    getNextToken: function(token, options) {
         var index = token._tokenIndex + 1;
-        return index < this._tree.tokens.length ? this._tree.tokens[index] : undefined;
+
+        if (index >= this._tokens.length) {
+            return undefined;
+        }
+
+        if (options && options.includeComments) {
+            return this._tokens[index];
+        }
+
+        do {
+            if (!this._tokens[index].isComment) {
+                return this._tokens[index];
+            }
+        } while (++index < this._tokens.length);
```

Comment:
I just ran the same thing you did and got:

master
```
real	0m9.353s
user	0m8.401s
sys	0m0.313s
```


jscs-dev-mdevils.autofixing-token-data
```
real	0m10.031s
user	0m8.752s
sys	0m0.415s
```

Which is a negligible difference. But then again, BOTH of our results are invalid because they are not rigorous done by a perf testing framework, which exist and we should integrate if we care about this sort of thing.